mne/cov.py

```diff
@@ -1751,6 +1768,10 @@ def prepare_noise_cov(
 
             dict(mag=1e12, grad=1e11, eeg=1e5)
     %(on_rank_mismatch)s
+    on_few_samples : str
```

Comment:
Only functions that *compute* a covariance should end up warning about the number of samples. Things like `prepare_noise_cov` and `regularize` shouldn't have any new public parameter. Under the hood they should pass `on_few_samples="ignore"` (or maybe nothing should need to be passed for it to work?)